api/mod/tests/conftest.py

```diff
@@ -30,3 +41,47 @@ def session() -> Generator:
     yield SETTINGS
 
     shutil.rmtree(SETTINGS.data_folder)
+
+
+@pytest.fixture(scope="session", autouse=True)
+def apply_migrations() -> Generator:
+    """Apply migrations to test database"""
+    config = Config(str(Path.cwd().joinpath("alembic.ini")))
+    config.set_main_option(
+        "script_location", str(Path.cwd().joinpath("mod").joinpath("alembic"))
+    )
+    config.set_main_option("sqlalchemy.url", SETTINGS.database_url)
+    alembic.command.upgrade(config, "head")
+    yield
+    alembic.command.downgrade(config, "base")
+
+
+@pytest.fixture(autouse=True)
+def db() -> scoped_session:
+    """
+    Provides a database session that is scoped to the test function
+    and is automatically rolled back after the test
+    """
+    connection = engine.connect()
+    transaction = connection.begin()
+    session = scoped_session(
+        sessionmaker(autocommit=False, autoflush=False, bind=connection)
+    )
+
+    def override_get_db() -> scoped_session:
+        return session
+
+    APP.dependency_overrides[get_db] = override_get_db
```

Comment:
Whatever this wild thing does :shrug: :sweat_smile:

Reply:
xD
It comes back to the `Depends` functionality: the `get_db` function is registered in fastapi as a dependency for the route handlers, and we override this dependency here to use a different function for obtaining the database session

Reply:
Rename to `db`